Rx/v2/src/rxcpp/operators/rx-debounce.hpp

```diff
@@ -140,13 +140,15 @@ struct debounce
             return std::function<void(const rxsc::schedulable&)>(selectedProduce.get());
         }
 
-        void on_next(T v) const {
+        template<typename U>
+        void on_next(U&& v) const {
+            auto vAsShared = std::make_shared<T>(std::forward<U>(v));
```

Comment:
an alloc per on_next hurts me :( 
could we just make the lambda mutable and move v instead of copying it?
`localState->value.reset(std::move(v));`